client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -109,11 +104,7 @@ export function Gutenboard() {
 									aria-label={ NO__( 'Onboarding screen content' ) }
 									tabIndex={ -1 }
 								>
-									<WritingFlow>
-										<ObserveTyping>
-											<BlockList className="gutenboarding-block-list" />
-										</ObserveTyping>
-									</WritingFlow>
```

Comment:
Will we need these wrappers in the future? If not 👍 

But if so, it might be good to leave them in and come up with an alternative, though I can't think of anything aside from overriding the CSS 🤷‍♂ 

```
.block-editor-writing-flow__click-redirect {
    cursor: text;
}
```

@sirreal - would you happen to have an inkling on the likelihood of having multiple blocks here? Thanks!

Reply:
I think if we want them in the future then we could use some component state to conditionally add them. At the moment Gutenboarding isn't in a writing type mode so we don't want all the cursor tracking. But then if we add a new step in the flow that transitions to something more editory we could conditionally render them.

Reply:
The structure of this was heavily inspired by the [Gutenberg playground](https://github.com/WordPress/gutenberg/blob/master/storybook/stories/playground/index.js).

> @sirreal - would you happen to have an inkling on the likelihood of having multiple blocks here?

Ideally, we have a smooth gradual transition from the current experience into the editor until there's a complete, functional editor in one, seamless flow.

That's a ways out, and whether or not we'll ever make it there is up in the air and involves other concerns.

I would like to have a confident understanding of what these components offer before deciding to remove them, this sounds like we don't to me:

> The `WritingFlow and `ObserveTyping` [components] seem to be used for communicating between blocks

I don't know what exactly they do and haven't located them in the editor component tree, I just know they came from the playground. If we can determine the above is accurate and doesn't provide any helpful functionality, it seems fine to remove them.

Reply:
> `<ObserveTyping />` is a component used in managing the editor's internal typing flag. When used to wrap content — typically the top-level block list — it observes keyboard and mouse events to set and unset the typing flag. The typing flag is used in considering whether the block border and controls should be visible. While typing, these elements are hidden for a distraction-free experience.

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/observe-typing/README.md

I think when we (🤞) seamlessly transition into a more editor like experience this is when we would add this into the component tree.

In saying that, it doesn't look like the text cursor is actually coming from this component (it's the other one) so I don't think it actually needs to be removed to fix this bug.

---

> **WritingFlow**
> Handles selection and navigation across blocks. This component should be
wrapped around BlockList.

https://developer.wordpress.org/block-editor/packages/packages-block-editor/#WritingFlow

From reading the code this handles things like changing which block is selected when in navigation mode, or when pressing DOWN when the caret is at the bottom of a text block.

Also something that's only needed when Gutenboarding starts to feel like an editor.

---

Maybe the ideal solution would be removing the text cursor from `<WritingFlow>` upstream because I don't see anything in that component that looks like click-here-to-start-typing functionality. Maybe the cursor belongs on `<BlockList>` or something.

Reply:
🙇 Thanks for the investigation. Please feel free to remove it if you think that's best 👍

Reply:
hmm I would think it is best to remove both for now because even though ObserveTyping is not causing any issue it's still dead code and we can always add it back in when we have a site editor as part of gutenboarding.